File: src/main/java/jndev/bonebot/BoneBot.java
package jndev.bonebot;

import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.JDABuilder;

import javax.imageio.ImageIO;
import javax.security.auth.login.LoginException;

/**
 * BoneBot is a simple discord bot for the ISUCF'V'MB Trombone discord
 *
 * @author JNDev (Jeremaster101)
 */
public class BoneBot {
    
    /**
     * create the bot and run it
     *
     * @param args arg 0 is the bot token
     * @throws LoginException when unable to log in to bot account
     */
    public static void main(String[] args) throws LoginException {
        System.setProperty("apple.awt.UIElement", "true");
        java.awt.Toolkit.getDefaultToolkit();
        
        if (args.length < 1) {
            System.out.println("You have to provide a token as first argument!");
            System.exit(1);
        }
        
        JDA jda = JDABuilder.createLight(args[0])
                .addEventListeners(new Listener())
                .build();
        
        Halloween halloween = new Halloween(jda);
        jda.addEventListener(halloween);
        
        ImageIO.setUseCache(false);
        Loader.loadData("responses.txt", Responder.responses);
        Loader.loadData("texts.txt", Meme.texts);
        Loader.loadData("statuses.txt", Status.statuses);
        Loader.loadData("reactions.txt", Reactor.reactions);
    
        Status.setStatus(jda);
    }
}

File: src/main/kotlin/jndev/bonebot/BoneBot.kt
package jndev.bonebot

import jndev.bonebot.config.Config
import jndev.bonebot.config.Loader
import jndev.bonebot.listener.Listener
import jndev.bonebot.modules.*
import net.dv8tion.jda.api.JDABuilder
import java.awt.Toolkit
import javax.imageio.ImageIO
import javax.security.auth.login.LoginException

/**
 * Main class, initializes bot, loads all data, and initializes modules
 *
 * @author Jeremy Noesen
 */
object BoneBot {
    /**
     * create the bot and run it
     *
     * @param args
     * @throws LoginException when unable to log in to bot account
     */
    @Throws(LoginException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        System.setProperty("apple.awt.UIElement", "true")
        Toolkit.getDefaultToolkit()
        ImageIO.setUseCache(false)
        Config.load()
        Loader.loadData("resources/commands.txt", Command.commands)
        Loader.loadData("resources/responses.txt", Responder.responses)
        Loader.loadData("resources/texts.txt", Meme.texts)
        Loader.loadData("resources/statuses.txt", Status.statuses)
        Loader.loadData("resources/reactions.txt", Reactor.reactions)
        val jda = JDABuilder.createLight(Config.botToken).addEventListeners(Listener()).build()
        Status.setStatus(jda)

//        Halloween halloween = new Halloween(jda);
//        jda.addEventListener(halloween);
    }
}